tests/sentry/search/events/test_builder.py

```diff
@@ -773,3 +846,193 @@ def get_granularity(start, end):
         start = datetime.datetime(2015, 5, 18, 10, 15, 1, tzinfo=timezone.utc)
         end = datetime.datetime(2015, 5, 19, 10, 15, 34, tzinfo=timezone.utc)
         assert get_granularity(start, end) == 60, "less than a minute"
+
+    def test_run_query(self):
+        self.store_metric(100, tags={"transaction": "foo_transaction"})
+        query = MetricsQueryBuilder(
+            self.params,
+            f"project:{self.project.slug}",
+            selected_columns=[
+                "transaction",
+                "p95(transaction.duration)",
+            ],
+        )
+        result = query.run_query("test_query")
+        assert len(result["data"]) == 1
+        assert result["data"][0] == {
+            "transaction": indexer.resolve("foo_transaction"),
+            "p95_transaction_duration": 100,
+        }
+
+    def test_run_query_multiple_tables(self):
+        self.store_metric(100, tags={"transaction": "foo_transaction"})
+        self.store_metric(
+            1,
+            metric=constants.METRICS_MAP["user"],
+            entity="metrics_sets",
+            tags={"transaction": "foo_transaction"},
+        )
+        query = MetricsQueryBuilder(
+            self.params,
+            f"project:{self.project.slug}",
+            selected_columns=[
+                "transaction",
+                "p95(transaction.duration)",
+                "count_unique(user)",
+            ],
+        )
+        result = query.run_query("test_query")
+        assert len(result["data"]) == 1
+        assert result["data"][0] == {
+            "transaction": indexer.resolve("foo_transaction"),
+            "p95_transaction_duration": 100,
+            "count_unique_user": 1,
+        }
+
+    def test_run_query_with_multiple_groupby_orderby_distribution(self):
+        self.setup_orderby_data()
+        query = MetricsQueryBuilder(
+            self.params,
+            f"project:{self.project.slug}",
+            selected_columns=[
+                "transaction",
+                "project",
+                "p95(transaction.duration)",
+                "count_unique(user)",
+            ],
+            orderby="-p95(transaction.duration)",
+        )
+        result = query.run_query("test_query")
+        assert len(result["data"]) == 2
+        assert result["data"][0] == {
+            "transaction": indexer.resolve("foo_transaction"),
```

Comment:
The result here is an integer, right? Should it be mapped back to the string for the response?

Reply:
TBH, still haven't decided if that makes more sense in a post-processing step, or here in the `run_query` function. Going to leave this for now and revisit when I connect this to an endpoint.